execution_engine/src/core/engine_state/era_validators.rs

```diff
@@ -0,0 +1,47 @@
+use casper_types::{
+    auction::{EraId, ValidatorWeights},
+    ProtocolVersion,
+};
+
+use crate::shared::newtypes::Blake2bHash;
+
+#[derive(Debug)]
+pub enum GetEraValidatorsResult {
+    /// A successful query operation returned validator weights for given era.
+    Success { validator_weights: ValidatorWeights },
+    /// Invalid state hash was used to make this request
+    RootNotFound,
+    /// Wrong era passed. There is no snapshot of validator weights for given era.
+    InvalidEra,
+    /// Indicates that queried value is missing, or has invalid type.
+    ValueError,
+}
```

Comment:
Wouldn't it be more idiomatic to make the last 3 variants a `GetEraValidatorsError` type, and use a `Result<ValidatorWeights, GetEraValidatorsError>` instead?

Reply:
Fixed